src/main/java/com/intel/genomicsdb/reader/GenomicsDBFeatureReader.java

```diff
@@ -47,6 +47,7 @@
  */
 public class GenomicsDBFeatureReader<T extends Feature, SOURCE> implements FeatureReader<T> {
     private String loaderJSONFile;
+    private String userProvidedQueryJSONFilename = null;
```

Comment:
You may want to rename it as 'queryJsonFile'. Both files (loader and query) are provided by the user.